tests/integration/ovs_test.py

```diff
@@ -15,106 +14,102 @@
 #
 # You should have received a copy of the GNU Lesser General Public License
 # along with this program. If not, see <https://www.gnu.org/licenses/>.
-#
 
 import pytest
-import yaml
 
-import libnmstate
+from libnmstate.schema import Interface
 from libnmstate.schema import InterfaceIPv4
+from libnmstate.schema import OVSBridge
+from libnmstate.schema import OVSBridgePortType
 from libnmstate.error import NmstateLibnmError
 
-from .testlib import statelib
-from .testlib.statelib import INTERFACES
+from .testlib import assertlib
+from .testlib.ovslib import add_port_to_bridge
+from .testlib.ovslib import create_bridge_subtree_state
+from .testlib.ovslib import ovs_bridge
+from .testlib.ovslib import ovs_interface
 
 
-OVS_BRIDGE_YAML_BASE = """
-interfaces:
-  - name: ovs-br0
-    type: ovs-bridge
-    state: up
-    bridge:
-      options:
-        fail-mode: ''
-        mcast-snooping-enable: false
-        rstp: false
-        stp: true
-"""
+@pytest.mark.xfail(
+    raises=NmstateLibnmError, reason='https://bugzilla.redhat.com/1724901'
+)
+def test_create_and_remove_ovs_bridge_with_min_desired_state():
+    with ovs_bridge('br1') as bridge_state:
+        assertlib.assert_state_match(bridge_state)
+    assertlib.assert_absent('br1')
 
 
 @pytest.mark.xfail(
     raises=NmstateLibnmError, reason='https://bugzilla.redhat.com/1724901'
 )
-def test_create_and_remove_ovs_bridge_with_a_system_port(eth1_up):
-    state = yaml.load(OVS_BRIDGE_YAML_BASE, Loader=yaml.SafeLoader)
-    state[INTERFACES][0]['bridge']['port'] = [
-        {'name': 'eth1', 'type': 'system'}
-    ]
-    libnmstate.apply(state)
-
-    setup_remove_ovs_bridge_state = {
-        INTERFACES: [
-            {'name': 'ovs-br0', 'type': 'ovs-bridge', 'state': 'absent'}
-        ]
-    }
-    libnmstate.apply(setup_remove_ovs_bridge_state)
-    state = statelib.show_only((state[INTERFACES][0]['name'],))
-    assert not state[INTERFACES]
+def test_create_and_remove_ovs_bridge_options_specified():
+    bridge = create_bridge_subtree_state(
+        {
+            OVSBridge.FAIL_MODE: '',
+            OVSBridge.MCAST_SNOOPING_ENABLED: False,
+            OVSBridge.RSTP: False,
+            OVSBridge.STP: True,
+        }
+    )
+
+    with ovs_bridge('br1', bridge) as bridge_state:
+        assertlib.assert_state_match(bridge_state)
+
+    assertlib.assert_absent('br1')
 
 
 @pytest.mark.xfail(
     raises=NmstateLibnmError, reason='https://bugzilla.redhat.com/1724901'
 )
-def test_create_and_remove_ovs_bridge_with_min_desired_state():
-    desired_state = {
-        INTERFACES: [{'name': 'ovs-br0', 'type': 'ovs-bridge', 'state': 'up'}]
-    }
-    libnmstate.apply(desired_state)
+def test_create_and_remove_ovs_bridge_with_a_system_port(eth1_up):
```

Comment:
For bridges we have an alias ‘port0_up’ instead of ‘eth1_up’.
And we fetch from it the actual name (see linux bridge tests).

Reply:
Weird and ugly. Done